### PR TITLE
WIP: Remove methods from const EVP_CIPHER structures

### PR DESCRIPTION
**Pull request by slontis**
Tests that use engines are current broken as they still run through the deprecated path.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/evp/evp_enc.c [hunk 1]
```diff
@@ -276,6 +276,7 @@ int EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
         case NID_rc5_ecb:
         case NID_rc5_cfb64:
         case NID_rc5_ofb64:
+        case NID_rc2_ecb:
```
Comment (levitte) [inline crypto/evp/evp_enc.c]: At some point, we should stop maintaining this absurd list, and instead simply try to fetch, and fall back to legacy when that fails.

Comment (slontis) [reply crypto/evp/evp_enc.c]: At some point there should be no legacy (so also no list ) :)

File: include/crypto/evp.h [hunk 2]
```diff
@@ -278,6 +278,51 @@ struct evp_cipher_st {
...
+IMPLEMENT_EVP_CIPHER_CONST(alg, nid##_##keybits##_cfb1, keybits, 1, 16, cfb1 ,CFB, flags)  \
+IMPLEMENT_EVP_CIPHER_CONST(alg, nid##_##keybits##_cfb8, keybits, 1, 16, cfb8, CFB, flags)
+
+#define IMPLEMENT_EVP_CIPHER_CONST_modes2(alg, nid, keybits, blocklen, ivlen, cbits, flags)   \
+IMPLEMENT_EVP_CIPHER_CONST2(alg, nid##_ecb, keybits, blocklen, 0, ecb, ECB, flags)            \
+IMPLEMENT_EVP_CIPHER_CONST2(alg, nid##_cbc, keybits, blocklen, ivlen, cbc, CBC, flags)        \
+IMPLEMENT_EVP_CIPHER_CONST2(alg, nid##_ofb##cbits, keybits, 1, ivlen, ofb, OFB, flags)        \
+IMPLEMENT_EVP_CIPHER_CONST2(alg, nid##_cfb##cbits, keybits, 1, ivlen, cfb##cbits, CFB, flags)
+
+
```
Comment (levitte) [inline include/crypto/evp.h]: Do we have *any* use of these macros outside of `crypto/evp/`?  If not, they could as well be placed in `crypto/evp/evp_local.h`

File: crypto/evp/evp_enc.c (hunk 1 above)
Comment (levitte) [reply crypto/evp/evp_enc.c]: Yes, absolutely...  but I'm thinking that *even now*, we could throw away that list.  It was a help before we got everything down, but at this point, I think the new `EVP_CIPHER` and `EVP_MD` have matured enough to make that change.

File: include/crypto/evp.h (hunk 2 above)
Comment (slontis) [reply include/crypto/evp.h]: updated

**Comment by levitte** [comment]
The CIs are having a fit...  which simply indicates that there are still corners in the code that depend on the presence of the legacy stuff.

This (from Travis) could be a starting point:

```
/home/travis/build/openssl/openssl/_build/util/shlib_wrap.sh /home/travis/build/openssl/openssl/_build/test/dtls_mtu_test => 139
not ok 1 - running dtls_mtu_test
#   Failed test 'running dtls_mtu_test'
#   at ../../_srcdist/test/recipes/80-test_dtls_mtu.t line 21.
# Looks like you failed 1 test of 1.
80-test_dtls_mtu.t ................. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
```

(139 is a SIGSEGV if I'm not mistaken)

**Comment by slontis** [comment]
Most of the 70- ones are related to engines ossltest being used. So this is a good test of engines :)
Some of the others are related to the 2 missing ciphers (one PR just merged + one more existing one to go)

**Comment by slontis** [comment]
Closing as this is not going to work for engines.

**Pull request closed without merging**